File: app/src/main/java/com/kevinmost/leddit/prefs/preferences/StringPreference.java
package com.kevinmost.leddit.prefs.preferences;

import com.kevinmost.leddit.prefs.LedditPreference;
import com.kevinmost.leddit.prefs.LedditPreferences;

import javax.inject.Inject;

public enum StringPreference implements LedditPreference<String> {
    REDDIT_TOKEN("Auth token"),
    ;

    @Inject
    LedditPreferences prefs;

    private final String key;

    private StringPreference(String key) {
        this.key = key;
    }

    @Override
    public String getValue() {
        return prefs.getSharedPreferences().getString(key, "");
    }

    @Override
    public void setValue(String value) {
        prefs.put(this, value);
    }

    @Override
    public String getKey() {
        return key;
    }
}

File: app/src/main/kotlin/com/kevinmost/leddit/preference/StringPreference.kt
package com.kevinmost.leddit.preference

import android.content.SharedPreferences

public enum class StringPreference private(key: String, defaultValue: String = ""):
        LedditPreference<String> {

    override protected var prefs: SharedPreferences? = null;
    override protected val key: String = key;
    override protected val defaultValue: String = defaultValue;

    override protected fun SharedPreferences.Editor.put(value: String): SharedPreferences.Editor {
        putString(key, value);
        return this;
    }
    override protected fun SharedPreferences.get(): String {
        return getString(key, defaultValue);
    }
}